app/user.go

```diff
@@ -1719,12 +1723,24 @@ func (a *App) AutocompleteUsersInChannel(teamId string, channelId string, term s
 	esInterface := a.Elasticsearch
 	license := a.License()
 	if esInterface != nil && *a.Config().ElasticsearchSettings.EnableAutocomplete && license != nil && *license.Features.Elasticsearch {
-		uchanIds, nuchanIds, err := a.Elasticsearch.SearchUsersInChannel(teamId, channelId, term, options)
+		listOfAllowedChannels, err := a.getListOfAllowedChannelsForTeam(teamId, options.ViewRestrictions)
 		if err != nil {
 			return nil, err
 		}
-		uchan = a.Srv.Store.User().GetProfileByIds(uchanIds, false)
-		nuchan = a.Srv.Store.User().GetProfileByIds(nuchanIds, false)
+		if len(listOfAllowedChannels) == 0 {
+			return &model.UserAutocompleteInChannel{}, nil
+		}
+		searchChannelId := channelId
+		if !strings.Contains(strings.Join(listOfAllowedChannels, "."), channelId) {
+			searchChannelId = "INVALID-CHANNEL-ID"
```

Comment:
This is a small trick to exclude the current channel if I haven't the permission to see the members of that channel. Was written with @mgdelacroix to avoid the limitations of the current elastic search implementation approach.

Reply:
I've been thinking about this and with the final interface, to avoid using this trick we can do:

```
uchanIds := []string{}
nuchanIds := []string{}
if !strings.Contains(strings.Join(listOfAllowedChannels, "."), channelId) {
	nuchanIds, err = a.Elasticsearch.SearchUsersInChannels(listOfAllowedChannels, term, options)
} else {
	uchanIds, nuchanIds, err = a.Elasticsearch.SearchUsersInChannelAndOutside(searchChannelId, listOfAllowedChannels, term, options)
}
if err != nil {
	return nil, err
}

...
```

It's a little longer but states more clearly the idea of "if your channel is not allowed, the autocompletion for channel users is an empty slice"

Reply:
Good point, I'll do it :) Thanks

Reply:
Done :)